src/main/java/net/starlark/java/eval/StarlarkList.java

```diff
@@ -286,7 +286,7 @@ public String toString() {
 
     // TODO(adonovan): reject unreasonably large n.
     int ni = n.toInt("repeat");
-    Object[] res = new Object[ni * size];
+    Object[] res = new Object[StarlarkInt.multiplyExact(ni, size, "repeat")];
```

Comment:
Let's reject excessive requests at the same time. We don't need the new helper functions to do that. Here's the logic (ported) from go.starlark.net:

```
// It's always possible to overeat in small bites but we'll
// try to stop someone swallowing the world in one gulp.
static final MAX_ALLOC = 1 << 30;
...

        // Inv: ni > 0, size > 0
        int sz = size * ni;
        if (sz < 0 || sz >= MAX_ALLOC) { // sz < 0 => overflow
                // Don't print sz.
                throw Starlark.errorf("excessive repeat (%d * %d elements)", size, ni);
        }
```

Reply:
> static final MAX_ALLOC = 1 << 30;

Does it make sense to limit size to 1G elements (4G/8G memory), while Java limit for array size is 2G-1 elements (8G/16G memory)? Seems to me like an insignificant reduction for additional complexity.

Reply:
1G should be enough for anyone, and I would prefer we fail fast with a clear error for accidental use than risk allocating an unreasonable amount of memory or hitting edge cases of the underlying allocator.  We can up the limit if users find it bothersome, but I don't think any actual user is even remotely close to it.